.gitlab-ci.yml

```diff
@@ -0,0 +1,210 @@
+  - python --version
+  - python3 --version
+
+##########################
+# Build Matrix Variables #
+##########################
+
+.buildtype_debug: &buildtype_debug
+  BUILD_TYPE: "Debug"
+
+.buildtype_release: &buildtype_release
+  BUILD_TYPE: "Release"
+
+.skylake: &skylake
+    SCHEDULER_PARAMETERS: "--nodes=1 --partition=skylake-gold,skylake-platinum"
+
+.power9: &power9
+    SCHEDULER_PARAMETERS: "--nodes=1 --partition=power9"
+
+#################
+# General Setup #
+#################
+
+.job: &job_def
+  tags:
+    - darwin-slurm-shared
+  variables:
+    SCHEDULER_PARAMETERS: "--nodes=1 --partition=skylake-gold,skylake-platinum"
+  only:
+    refs:
+      - master
+      - merge_requests
+  except:
+    refs:
+      - schedules
+
+.artifacts: &artifacts_def
+  artifacts:
+    name: "singularity-eos-${CI_COMMIT_REF_SLUG}-${CI_JOB_NAME}-${CI_JOB_ID}"
+    paths:
+      - ${CI_PROJECT_DIR}/build
+      - ${CI_PROJECT_DIR}/spack
+      - ${CI_PROJECT_DIR}/spack_env
+    when: always
+    expire_in: 3 hours
+  dependencies: []
+
+#######################
+# Job Script Template #
+#######################
+
+.clang-format:
+  stage: static_analysis
+  variables:
+    GIT_SUBMODULE_STRATEGY: none
+  before_script:
+    - echo "Format Checking..."
+  script:
+    - cd ${CI_PROJECT_DIR}
+    - module purge
+    - module load clang/${SINGULARITY_EOS_CLANG_FORMAT_VERSION}
+    - clang-format -version
+    - find ${CI_PROJECT_DIR} -regex '.*\.\(cpp\|hpp\)' | xargs clang-format -style=file -i && git diff --exit-code --ignore-submodules
+
+.sphinx-doc:
+  stage: static_analysis
+  variables:
+    GIT_SUBMODULE_STRATEGY: none
+  script:
+    - cd ${CI_PROJECT_DIR}
+    - mkdir sphinx_doc_env
+    - spack env create --without-view -d ${CI_PROJECT_DIR}/sphinx_doc_env
+    - spack env activate -d ${CI_PROJECT_DIR}/sphinx_doc_env
+    - spack install --show-log-on-error py-sphinx%gcc@${SINGULARITY_EOS_GCC_VERSION}
+    - spack install --show-log-on-error py-sphinx-rtd-theme%gcc@${SINGULARITY_EOS_GCC_VERSION}
+    - spack module tcl refresh -y
+    - spack env loads -r
+    - source ${CI_PROJECT_DIR}/sphinx_doc_env/loads
+    - cd ${CI_PROJECT_DIR}/doc/sphinx
+    - make html
+    - rm -rf ${CI_PROJECT_DIR}/public
+    - mv _build/html ${CI_PROJECT_DIR}/public
+  artifacts:
+    paths:
+      - public
+  only:
+    - master
+
+.test:
+  stage: build_n_test
+  extends:
+    - .job
+  script:
+    - python --version
+    - python3 --version
+    - mkdir -p build
+    - cd build
+    - |
+      cmake -DSINGULARITY_USE_EOSPAC=ON \
+            -DSINGULARITY_USE_HDF5=ON \
+            -DSINGULARITY_BUILD_SESAME2SPINER=ON \
+            -DSINGULARITY_BUILD_STELLARCOLLAPSE2SPINER=ON \
+            -DSINGULARITY_BUILD_TESTS=ON \
+            -DSINGULARITY_TEST_SESAME=ON \
+            -DSINGULARITY_TEST_STELLAR_COLLAPSE=ON \
+            ..
+    - make -j
+    - ./sesame2spiner/sesame2spiner -s materials.sp5 ../sesame2spiner/examples/unit_tests/*.dat
```

Comment:
Doesn't the spiner file need to get moved to the `singularity-eos-data` directory in order for certain tests to run? Will `make test` trigger a re-build such that the spiner file is copied into the necessary build directory and dependent tests are built and linked?

Reply:
I changed the path pointing to the file in the `unit_tests` file. Now it points one directory up:
```C++
const std::string eosName = "../materials.sp5";
```

Reply:
Will `make test` re-check for the existence of that file and then proceed to build the eos test or will it just run the tests as built from the previous make command (which would not include the eos test since the needed file didn't exist at the time the make command was run).

Reply:
`make test` won't check for the existence of the sp5 file. But the `eos_unit_tests` binary will be built regardless, as the gold file is no longer listed as a build dependency. So if the commands are run in this order, it'll all work.